app/controllers/application_controller.rb

```diff
@@ -1,2 +1,48 @@
+# frozen_string_literal: true
+
 class ApplicationController < ActionController::Base
+  protect_from_forgery with: :exception
+  helper_method :logged_in?
+
+  def log_in(user)
+    session[:user_id] = user.id
+  end
```

Comment:
For this method, you should generate the remember token and update the signed-in user with that token.